tracepoint/src/native.rs

```diff
@@ -421,15 +412,58 @@ impl TracepointState {
 
     /// Registers this tracepoint.
     ///
-    /// Requires: this tracepoint is not currently registered.
+    /// Requires: this `TracepointState` is not currently registered.
+    ///
+    /// Returns 0 for success, error code (e.g. EACCES, ENOENT) for error. The error code
+    /// is usually ignored in retail scenarios but may be helpful during development to
+    /// understand behavior or track down issues.
+    ///
+    /// `_name_args` is the tracepoint definition in the format
+    /// `Name[ FieldDef1[;FieldDef2...]]`. For example:
+    ///
+    /// - `MyTracepoint1`
+    /// - `MyTracepoint2 u32 Field1`
+    /// - `MyTracepoint3 u32 Field1;char Field2[20]`
     ///
     /// # Safety
     ///
-    /// The tracepoint must be unregistered before it is deallocated. Note that it will
-    /// unregister itself when dropped, so this is only an issue if the tracepoint is
-    /// not dropped before it is deallocated, as might happen for a static variable in a
-    /// shared library that gets unloaded.
+    /// The tracepoint must be unregistered before it is deallocated. `TracepointState`
+    /// will unregister itself when dropped, so this is only an issue if the tracepoint
+    /// is not dropped before it is deallocated. This might happen for a static variable
+    /// in a shared library that gets unloaded.
     pub unsafe fn register(self: Pin<&Self>, _name_args: &ffi::CStr) -> i32 {
+        return self.register_with_flags(_name_args, 0);
+    }
+
+    /// Advanced: Registers this tracepoint using the specified `user_reg` flags.
+    ///
+    /// Requires: this `TracepointState` is not currently registered.
+    ///
+    /// Returns 0 for success, error code (e.g. EACCES, ENOENT) for error. The error code
+    /// is usually ignored in retail scenarios but may be helpful during development to
+    /// understand behavior or track down issues.
+    ///
+    /// `_name_args` is the tracepoint definition in the format
+    /// `Name[ FieldDef1[;FieldDef2...]]`. For example:
+    ///
+    /// - `MyTracepoint1`
+    /// - `MyTracepoint2 u32 Field1`
+    /// - `MyTracepoint3 u32 Field1;char Field2[20]`
+    ///
+    /// `_flags` is normally `0`, but may also be set to a `user_reg` flag such as
+    /// `USER_EVENT_REG_PERSIST`.
+    ///
+    /// # Safety
+    ///
+    /// The tracepoint must be unregistered before it is deallocated. `TracepointState`
+    /// will unregister itself when dropped, so this is only an issue if the tracepoint
+    /// is not dropped before it is deallocated. This might happen for a static variable
+    /// in a shared library that gets unloaded.
+    pub unsafe fn register_with_flags(
```

Comment:
Note: New method `register_with_flags` added so there is a way for a user to specify registration flags. This isn't currently a common logging scenario, but flags are needed for a `tracepoint-register` tool.